pycram/src/pycram/local_transformer.py

```diff
@@ -0,0 +1,368 @@
+                robot_pose = BulletWorld.robot.get_position_and_orientation()
+            self.publish_robot_pose(helper.ensure_pose(robot_pose))
+
+    def update_robot(self):
+        self.update_robot_state()
+        self.update_robot_pose()
+
+    def update_objects(self):
+        if BulletWorld.current_bullet_world:
+            for obj in list(BulletWorld.current_bullet_world.objects):
+                if obj.name in robot_description.i.name or obj.name == "floor" or obj.name == "kitchen":
+                    continue
+                else:
+                    published = local_transformer.publish_object_pose(obj.name,
+                                                                      obj.get_position_and_orientation())
+                if not published:
+                    logerr("(publisher) Could not publish given pose of %s.", obj.name)
+
+    def update_static_odom(self):
+        self.publish_pose(self.map_frame, self.projection_namespace + '/' + self.odom_frame,
+                          [[0, 0, 0], [0, 0, 0, 1]], static=True)
+
+    def update_from_btr(self):
+        # Update static odom
+        self.update_static_odom()
+        # Update pose and state of robot
+        self.update_robot()
+        # Update pose of objects which are possibly attached on the robot
+        self.update_objects()
+
+    ############################################################################################################
+    ######################################### Query Local Transformer ##########################################
+    ############################################################################################################
+
+    def tf_pose_transform(self, source_frame, target_frame, pose, time=None):
+        tf_pose = helper.ensure_pose(pose)
+        tf_time = time if time else self.local_transformer.getLatestCommonTime(source_frame, target_frame)
+        tf_pose_stamped = PoseStamped(Header(0, tf_time, source_frame), tf_pose)
+        return self.tf_pose_stamped_transform(target_frame, tf_pose_stamped)
+
+    def tf_pose_stamped_transform(self, target_frame, pose_stamped):
+        return helper.pose_stamped2tuple(self.local_transformer.transformPose(target_frame, pose_stamped))
+
+    def tf_transform(self, source_frame, target_frame, time=None):
+        tf_time = time if time else self.local_transformer.getLatestCommonTime(source_frame, target_frame)
+        return self.local_transformer.lookupTransform(source_frame, target_frame, tf_time)
+
+    ############################################################################################################
+    ####################################### Publish to Local Transformer ###########'###########################
+    ############################################################################################################
+
+    def publish_pose(self, frame_id, child_frame_id, pose, static=None):
+        if frame_id is not child_frame_id:
```

Comment:
Be careful when using the "is" operator because it is not equivalent to the "==" operator. Please check https://realpython.com/courses/python-is-identity-vs-equality/